tabular/src/autogluon/tabular/configs/config_helper.py

```diff
@@ -0,0 +1,357 @@
+        Appends DatetimeFeatureGenerator() to the generator group.
+        """
+        self.config['enable_datetime_features'] = value
+        return self
+
+    def enable_text_special_features(self, value: bool = True) -> FeatureGeneratorBuilder:
+        """
+        Whether to use 'object' features identified as 'text' features to generate 'text_special' features such as word count, capital letter ratio, and symbol counts.
+        Appends TextSpecialFeatureGenerator() to the generator group.
+        """
+        self.config['enable_text_special_features'] = value
+        return self
+
+    def enable_text_ngram_features(self, value: bool = True) -> FeatureGeneratorBuilder:
+        """
+        Whether to use 'object' features identified as 'text' features to generate 'text_ngram' features.
+        Appends TextNgramFeatureGenerator(vectorizer=vectorizer) to the generator group.
+        """
+        self.config['enable_text_ngram_features'] = value
+        return self
+
+    def enable_raw_text_features(self, value: bool = True) -> FeatureGeneratorBuilder:
+        """
+        Whether to keep the raw text features.
+        Appends IdentityFeatureGenerator(infer_features_in_args=dict(required_special_types=['text'])) to the generator group.
+        """
+        self.config['enable_raw_text_features'] = value
+        return self
+
+    def enable_vision_features(self, value: bool = True) -> FeatureGeneratorBuilder:
+        """
+        [Experimental]
+        Whether to keep 'object' features identified as 'image_path' special type. Features of this form should have a string path to an image file as their value.
+        Only vision models can leverage these features, and these features will not be treated as categorical.
+        Note: 'image_path' features will not be automatically inferred. These features must be explicitly specified as such in a custom FeatureMetadata object.
+        Note: It is recommended that the string paths use absolute paths rather than relative, as it will likely be more stable.
+        """
+        self.config['enable_vision_features'] = value
+        return self
+
+    def vectorizer(self, value: BaseEstimator) -> FeatureGeneratorBuilder:
+        """
+        sklearn CountVectorizer object to use in TextNgramFeatureGenerator.
+        Only used if `enable_text_ngram_features=True`.
+        """
+        self.config['vectorizer'] = value
+        return self
+
+    def text_ngram_params(self, value: bool = True) -> FeatureGeneratorBuilder:
+        """
+        Appends TextNgramFeatureGenerator(vectorizer=vectorizer, text_ngram_params) to the generator group. See text_ngram.py for valid parameters.
+        """
+        self.config['text_ngram_params'] = value
+        return self
+
+    def build(self) -> ConfigBuilder:
+        self.parent.config['feature_generator'] = AutoMLPipelineFeatureGenerator(**self.config)
+        return self.parent
+
+
+class ConfigBuilder:
```

Comment:
re 'intellisense assistance', doesn't this already occur when calling `.fit` in an IDE? I suppose the benefit is that we can split up the documentation into manageable chunks?

Reply:
It won't give help on kwargs.
Couple other benefits: 1) parameter-level doc strings (there's a lot of good information in `fit` docstring, but you have to dig it to find specific information on a parameter; 2) can do more complex setups/cross-checks on top of existing parameters. Technically things like presets can live in builders.